java-frontend/src/main/java/org/sonar/java/se/FlowComputation.java

```diff
@@ -331,7 +331,7 @@ public int hashCode() {
         return Optional.empty();
       }
       SymbolicValue peekValue = edge.child.programState.peekValue();
-      if (peekValue instanceof SymbolicValue.ExceptionalSymbolicValue) {
+      if (peekValue instanceof SymbolicValue.ExceptionalSymbolicValue && edge.parent.programPoint.syntaxTree() != null) {
```

Comment:
I would like a comment here. Especially if we plan to remove this once we fix clearing of exceptional SV when reaching `finally` block.